include/pybind11/cast.h

```diff
@@ -310,9 +335,7 @@ template <typename type> class type_caster<std::reference_wrapper<type>> : publi
 
 
 template <typename T>
-struct type_caster<
-    T, typename std::enable_if<std::is_integral<T>::value ||
-                               std::is_floating_point<T>::value>::type> {
+struct type_caster<T, typename std::enable_if<std::is_arithmetic<T>::value>::type> {
```

Comment:
It wasn't removed, just changed to is_arithmetic, which is true for both integral and floating point types.